File: participant-manager-module/participant-manager-service/src/main/java/com/google/cloud/healthcare/fdamystudies/beans/ParticipantRegistryDetail.java
/*
 * Copyright 2020 Google LLC
 *
 * Use of this source code is governed by an MIT-style
 * license that can be found in the LICENSE file or at
 * https://opensource.org/licenses/MIT.
 */

package com.google.cloud.healthcare.fdamystudies.beans;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.google.cloud.healthcare.fdamystudies.common.ErrorCode;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@JsonInclude(Include.NON_NULL)
@Component
@Scope(value = "prototype")
@ToString
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class ParticipantRegistryDetail extends BaseResponse {

  private String studyId;

  private String customStudyId;

  private String studyType;

  private Integer targetEnrollment;

  private String studyName;

  private String appId;

  private String customAppId;

  private String appName;

  private String siteId;

  private String customLocationId;

  private String locationName;

  private Integer locationStatus;

  private Integer sitePermission;

  private Integer siteStatus;

  private List<ParticipantDetail> registryParticipants = new ArrayList<>();

  private Map<String, Long> countByStatus;

  public ParticipantRegistryDetail(ErrorCode errorCode) {
    super(errorCode);
  }
}

File: participant-manager/src/app/shared/participant-registry-detail.ts
import {RegistryParticipant} from './participant';
import {Permission} from './permission-enums';
import {StudyType} from './enums';

export interface ParticipantRegistryDetail {
  studyId: string;
  customStudyId: string;
  studyType: StudyType;
  targetEnrollment: number;
  studyName: string;
  appId: string;
  customAppId: string;
  appName: string;
  siteId?: string;
  customLocationId?: string;
  locationName?: string;
  locationStatus?: string;
  sitePermission?: Permission;
  openStudySitePermission?: Permission;
  siteStatus?: number;
  registryParticipants: RegistryParticipant[];
  countByStatus: CountByStatus;
  status?: string;
}
export interface CountByStatus {
  // eslint-disable-next-line @typescript-eslint/naming-convention
  A?: number;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  D?: number;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  E?: number;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  I?: number;
  // eslint-disable-next-line @typescript-eslint/naming-convention
  N?: number;
}
